### ISSUE TITLE
[Feature Request]: workflows are missing a ExecutionStoppedListener

### ISSUE THREAD

**Issue by hansva** [apache/hop#3715]
### What would you like to happen?

Add ExecutionStoppedListener to Workflows too in analogy with pipelines

### Issue Priority

Priority: 2

### Issue Component

Component: API

**Comment by hansva** [comment]
.take-issue

**Comment by nadment** [comment]
I'd also be tempted to standardize methods between IPipelineEngine and IWorkflowEngine.
![image](https://github.com/apache/hop/assets/39568358/38f31bdb-c46f-4aa8-8252-f922d04d26f5)

Or a variant but even more API break: 
[add][remove]PipelineExecutionStartedListener
[fire]PipelineExecutionStartedListener
[add][remove]WorkflowExecutionStartedListener
[fire]WorkflowExecutionStartedListener

**Comment by hansva** [comment]
I don't think we have to wait for a 3.0 version for this. You might have also noticed with your work that the pipeline finished listener is triggered too soon, not all transforms have finished when it is triggered

**Comment by nadment** [comment]
I've seen bug #3675 but haven't seen it in action yet. I'll keep an eye out for it.
